### PR TITLE
Refactor Content Filtering

### PR DESCRIPTION
**Pull request by vianxnguyen**
## Overview

Refactor content filter tab bar used multiple times throughout codebase.

## Changes Made

Create ContentFilterBarView & use for filtering in bookmarks, publications, and search results.

## Test Coverage
Simulator.

https://user-images.githubusercontent.com/46759999/226811568-fdbbcc5d-1dec-432b-be81-d51d91ba5456.mp4

### CODE REVIEW

**Review by vinnie4k — APPROVED** [review]
Looks good!

File: Volume/Supporting Views/ContentFilterBarView.swift [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+                    width: Constants.magazineTabWidth
+                ) : nil
+            ].compactMap { $0 }
+        )
+    }
+    
+    private struct Constants {
+        static let articleTabWidth: CGFloat = 80
+        static let magazineTabWidth: CGFloat = 106
+
```
Comment (ReadeP) [inline Volume/Supporting Views/ContentFilterBarView.swift:37]: remove newline

**Review by ReadeP — APPROVED** [review]
LGTM!

File: Volume/Supporting Views/ContentFilterBarView.swift [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+                    title: "Articles",
+                    tab: .articles,
+                    width: Constants.articleTabWidth
+                ) : nil,
+                showMagazineTab ? SlidingTabBarView.Item(
+                    title: "Magazines",
+                    tab: .magazines,
+                    width: Constants.magazineTabWidth
+                ) : nil
+            ].compactMap { $0 }
```
Comment (dishanest) [inline Volume/Supporting Views/ContentFilterBarView.swift:30]: Nice!

File: Volume/Supporting Views/SearchResultsList.swift [hunk 3]
```diff
@@ -64,19 +64,16 @@ struct SearchResultsList: View {
...
+            ContentFilterBarView(selectedTab: $selectedTab, showMagazineTab: false)
+            RefreshableScrollView(onRefresh: { done in
+                if case let .results(articles) = sectionStates.articles {
+                    sectionStates.articles = .reloading(articles)
+                }
+                
+                sectionStates.articles = .loading
+                
+                fetchContent(done)
             }) {
```
Comment (dishanest) [inline Volume/Supporting Views/SearchResultsList.swift:77]: Use trailing closure syntax here. e.g. https://www.hackingwithswift.com/swift/5.3/multiple-trailing-closures

**Review by dishanest — APPROVED** [review]
Dope! One small syntax change but feel free to merge after you consider it.

**Pull request closed without merging**